### DISCUSSION TITLE
Long-lasting config setup

### DISCUSSION THREAD

**Discussion by jordimasmi** [cyberman54/ESP32-Paxcounter#797] [Q&A]
First of all, congratulations for the project.
I’m doing a proof of concept, to know the number of people who pass through a certain area of a natural park.

I am using a T-Beam (2400mAh - 18650) + PaxCounter 2.2.0, collecting data from Wifi devices, and sending the information to TTN. It works fine, but I don't get much autonomy.

I only send data from Wi-Fi devices and battery level.  BLE off, GPS off ...
All other parameters by default.

How much autonomy is it supposed to have? 
I don't get more than a day, if I don't put an auxiliary battery or solar panel …

Thanks in advance !

**Comment by cyberman54** [comment]
Wifi or BLE sniffing is a power consuming technology, if you want the radio always on for sniffing. For this use case you will need ~0,7W, and thus you must generate and deliver this power somehow to the device. If you can't, the only way is reducing power consumption by setting up a compromise, what means the device is not sniffing all the time.

Thus, you need a duty cycle, what means switching the sniffing on and off by a certain scheme.

You may use current paxcounter v2.5.0 in development branch to setup a duty cycle, based on power saving by deep sleep. You can configure a wake up either by time, or by an external sensor.

E.g. sniffing for 30 seconds, then sleeping for 2 minutes, resulting in a duty cycle of 1/4, will result to ~4 days of autonomy, if you have 1 day without sleep.

**Reply by jordimasmi** [reply]
Thank you very much for the answers, to see what I can improve.

**Comment by jordimasmi** [comment]
Another question .... 

When a device is detected (Wifi or Ble) ... and it is continuously scanned, and the information is sent every minute ... 
if the same Wifi / Ble is still in the radius of action ... it is sent/counted again ? or is it not accounted for? 
perhaps it will be counted again if it has left and re-entered the area?

It is not clear to me in the configuration what is the behavior of the count ...  if the devices are repeated or not.

Thanks in advance !

**Reply by cyberman54** [reply]
The behaviour depends on settings in [paxcounter.conf](https://github.com/cyberman54/ESP32-Paxcounter/blob/master/src/paxcounter_orig.conf#L15):

```
#define SENDCYCLE                 30      // payload send cycle [seconds/2], 0 .. 255
#define COUNTERMODE                0      // 0=cyclic, 1=cumulative, 2=cyclic confirmed

```
In countermode 0 and 2: all generated anonymous device identifiers are cleared after each sendcycle. Thus, same devices still present, will be counted again once, regardless if they disappear and reappear while still in same sendcycle.

In countermode 1: anonymous device identifiers are never cleared, thus there will be collisions, after the identifier memory reaches it's limit, which is 65535 different identifiers. Thus, same devices still present, will not be counted again after they were first seen.

The countermode can be switched during runtime by remote command 0x02:

```
0 = cyclic unconfirmed, mac counter reset after each wifi scan cycle, data is sent only once [default]
1 = cumulative counter, mac counter is never reset
2 = cyclic confirmed, like 0 but data is resent until confirmation by network received

```

PS: If you need to reduce power consumption, switch off Wifi sniffing and use BLE only.
